applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.h

```diff
@@ -0,0 +1,449 @@
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     * @param pProperties Pointer to the element's properties
+     */
+    QSVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties);
+
+    /// Destructor.
+    ~QSVMSDEMCoupled() override;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    /// Create a new element of this type
+    /**
+     * Returns a pointer to a new QSVMSDEMCoupled element, created using given input
+     * @param NewId the ID of the new element
+     * @param ThisNodes the nodes of the new element
+     * @param pProperties the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Create(IndexType NewId,
+                            NodesArrayType const& ThisNodes,
+                            Properties::Pointer pProperties) const override;
+
+    /// Create a new element of this type using given geometry
+    /**
+     * Returns a pointer to a new FluidElement element, created using given input
+     * @param NewId the ID of the new element
+     * @param pGeom a pointer to the geomerty to be used to create the element
+     * @param pProperties the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+     Element::Pointer Create(IndexType NewId,
+                             GeometryType::Pointer pGeom,
+                             Properties::Pointer pProperties) const override;
+
+
+    /**
+     * clones the selected element variables, creating a new one
+     * @param NewId: the ID of the new element
+     * @param ThisNodes: the nodes of the new element
+     * @param pProperties: the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    void Calculate(
+        const Variable<double>& rVariable,
+        double& rOutput,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void Calculate(const Variable<array_1d<double, 3>>& rVariable,
+                   array_1d<double, 3>& rOutput,
+                   const ProcessInfo& rCurrentProcessInfo) override;
+
+    void Calculate(
+        const Variable<Vector >& rVariable,
+        Vector& Output,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void Calculate(
+        const Variable<Matrix >& rVariable,
+        Matrix& Output,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+
+    void EquationIdVector(EquationIdVectorType& rResult,
+                          ProcessInfo& rCurrentProcessInfo) override;
+
+    void GetDofList(DofsVectorType& rElementalDofList,
+                    ProcessInfo& rCurrentProcessInfo) override;
+
+    double FilterWidth();
+
+    double FilterWidth(const BoundedMatrix<double, NumNodes, Dim >& DN_DX);
```

Comment:
I think these two can be `private` (or at least `protected`). Please check this for the other methods that are not `override`-specified.